### PR TITLE
CR-1206432 : Incorrect Help printed in xrt-smi when using incorrect option

### PR DESCRIPTION
**Pull request by aktondak**
#### Problem solved by the commit
This fixes CR 1206432 : Incorrect Help printed in xrt-smi validate and xrt-smi examine when using incorrect option

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
[CR 1206432](https://jira.xilinx.com/browse/CR-1206432). Discovered through internal testing

#### How problem was solved, alternative solutions (if any) and why they were rejected
The problem was solved by disabling cmdline-options-validation in the lower level API (XBU::process_arguments) and adding option validation responsibility to top level (validate subcommand and examine sub-command) so that the correct help is printed in respective cases of failures.

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
validate behavior on incorrect option :

```Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi validate --wrong
ERROR: Unrecognized arguments:
  --wrong


COMMAND: validate

DESCRIPTION: Validates the given device by executing the platform's validate executable.

USAGE: xrt-smi validate [--help] [-d arg] [-f arg] [-o arg] [-p arg] [-r arg] [--param arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --param            - Extended parameter for a given test. Format: <test-name>:<key>:<value>
  -p, --path         - Path to the directory containing validate xclbins
  --help             - Help to use this sub-command
  -r, --report       - The type of report to be produced. Reports currently available are:
                         all                  - All applicable validate tests will be executed
                                                (default)
                         cmd-chain-latency    - Run end-to-end latency test using command chaining
                         cmd-chain-throughput - Run end-to-end throughput test using command
                                                chaining
                         df-bw                - Run bandwidth test on data fabric
                         gemm                 - Measure the TOPS value of GEMM operations
                         latency              - Run end-to-end latency test
                         quick                - Only the first 4 tests will be executed
                         tct-all-col          - Measure average TCT processing time for all
                                                columns
                         tct-one-col          - Measure average TCT processing time for one column
                         throughput           - Run end-to-end throughput test

EXTENDED KEYS:
  dma                - block-size:<value> - Memory transfer size (bytes)


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
ERROR: operation canceled
```

Examine behavior on incorrect option :
```
Z:\Repos\XRT-MCDM-FORK\XRT-MCDM\build\WRelease\xilinx\xrt>xrt-smi examine --wrong**
ERROR: Unrecognized arguments:
  --wrong


COMMAND: examine

DESCRIPTION: This command will 'examine' the state of the system/device and will generate a report of interest
               in a text or JSON format.

USAGE: xrt-smi examine [--help] [-d arg] [-f arg] [-o arg] [-r arg]

OPTIONS:
  -d, --device       - The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest
  -f, --format       - Report output format. Valid values are:
                         JSON        - Latest JSON schema
                         JSON-2020.2 - JSON 2020.2 schema
  -o, --output       - Direct the output to the given file
  --help             - Help to use this sub-command
  -r, --report       - The type of report to be produced. Reports currently available are:
                         aie-partitions - AIE partition information
                         all            - All known reports are produced
                         host           - Host information
                         platform       - Platforms flashed on the device
                         telemetry      - Telemetry data for the device


GLOBAL OPTIONS:
  --verbose          - Turn on verbosity
  --batch            - Enable batch mode (disables escape characters)
  --force            - When possible, force an operation
ERROR: operation canceled
```
#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by sonals** [comment]
ok to test

File: src/runtime_src/core/tools/common/SubCmdExamineInternal.cpp [hunk 1]
```diff
@@ -103,8 +103,25 @@ SubCmdExamineInternal::execute(const SubCmdOptions& _options) const
...
+    if (!unrecognized_options.empty())
+    {
+      std::string error_str;
+      error_str.append("Unrecognized arguments:\n");
+      for (const auto& option : unrecognized_options)
+        error_str.append(boost::str(boost::format("  %s\n") % option));
+      throw boost::program_options::error(error_str);
+    }
+  }
+  catch (boost::program_options::error& e)
```
Comment (sonals) [inline src/runtime_src/core/tools/common/SubCmdExamineInternal.cpp:119]: const?

File: src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp [hunk 2]
```diff
@@ -521,8 +521,25 @@ void
...
+    if (!unrecognized_options.empty())
+    {
+      std::string error_str;
+      error_str.append("Unrecognized arguments:\n");
+      for (const auto& option : unrecognized_options)
+        error_str.append(boost::str(boost::format("  %s\n") % option));
+      throw boost::program_options::error(error_str);
+    }
+  }
+  catch (boost::program_options::error& e)
```
Comment (sonals) [inline src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp:537]: const?

**Review by rchane — APPROVED** [review]
Should be good to go after resolving Sonal's comments!

File: src/runtime_src/core/tools/common/SubCmdExamineInternal.cpp (hunk 1 above)
Comment (aktondak) [reply src/runtime_src/core/tools/common/SubCmdExamineInternal.cpp]: There is a top level catch in **SubCmd::process_arguments** that used to catch this earlier. But that catch printed a generic help (ryzenAI + Alveo) which was incorrect. Now this dedicated catch will catch this and print specific help based on the device. 
The const part is fixed now.

**Review by sonals — APPROVED** [review]
It would be best to reorganize the code to remove the code duplication in a future PR.

**Pull request merged**